install/troubleshooting_unused_drives_on_storage_nodes.md

```diff
@@ -0,0 +1,252 @@
+
+## Expected number of OSDs
+
+The following table shows the expected number of OSDs for every storage NCN.
+
+| Hardware Manufacturer | OSD Drive Count per Storage NCN (not including OS drives) |
+| :-------------------: | :---------------------------------------: |
+| GigaByte              | 12 |
+| HPE                   | 8  |
+
+The expected total number of OSDs on the system is determined by taking the number in the second column of the above
+table and multiplying it by the number of storage NCNs in the system. That is:
+
+`total_osds` = `(Number of utility storage nodes)` `*` `(OSD drive count per storage NCN)`
+
+## Automatic check
+
+1. Record the hardware type of the storage NCNs.
+
+    ```bash
+    HWTYPE=$(ipmitool mc info | grep "^Manufacturer Name" | sed 's/^Manufacturer Name[[:space:]][[:space:]]*:[[:space:]][[:space:]]*//')
+    echo "${HWTYPE}"
+    ```
+
+    Example output on a system with HPE hardware:
+
+    ```text
+    Hewlett Packard Enterprise
+    ```
+
+1. Record the number of storage NCNs on the system.
+
+    Be sure to set the `NUM_STORAGE_NCNS` variable to the actual number of storage NCNs on the system.
+
+    ```bash
+    NUM_STORAGE_NCNS=x
+    ```
+
+1. Execute the automatic check.
+
+    ```bash
+    /opt/cray/tests/install/ncn/scripts/python/check_for_unused_drives.py "${NUM_STORAGE_NCNS}" "${HWTYPE}"
+    ```
+
+    The final line of output will report whether or not the number of OSDs found matches what is expected.
+
+    Example of successful output on system with five HPE storage nodes:
+
+    ```text
+    2022-09-20 16:00:44.651 DEBUG    Parsing command line arguments: ['/opt/cray/tests/install/ncn/scripts/python/check_for_unused_drives.py', '5', 'Hewlett Packard Enterprise']
+    2022-09-20 16:00:44.652 INFO     Based on hardware type (hpe) and number of storage nodes (5): min_expected_osds = 40, max_expected_osds = 40
+    2022-09-20 16:00:44.652 DEBUG    Loading Ceph
+    2022-09-20 16:00:44.657 DEBUG    Connecting to Ceph
+    2022-09-20 16:00:44.664 INFO     Running ceph osd stat command
+    2022-09-20 16:00:44.671 INFO     Command return code = 0
+    2022-09-20 16:00:44.671 DEBUG    Decoding command output to string:
+    {
+        "epoch": 2634,
+        "num_osds": 40,
+        "num_up_osds": 40,
+        "osd_up_since": 1663023364,
+        "num_in_osds": 40,
+        "osd_in_since": 1659470908,
+        "num_remapped_pgs": 0
+    }
+    
+    2022-09-20 16:00:44.671 DEBUG    Decoding command output from JSON into object
+    2022-09-20 16:00:44.671 DEBUG    Extracting number of OSDs from object
+    2022-09-20 16:00:44.671 INFO     num_osds = 40
+    2022-09-20 16:00:44.671 INFO     SUCCESS -- number of OSDs found matches expectations based on hardware type
+    ```
+
+If the number of OSDs does not match what is expected, then proceed to [Manual checks and remediation](#manual-checks-and-remediation).
+
+## Manual checks and remediation
+
```

Comment:
We should add some sort of statement here about the differences in the options below.

Reply:
Done. Please re-review when you have the opportunity.